smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -112,6 +117,7 @@ contract MixinTransfer is
     onlyIfAlive
     hasValidKey(_from)
     onlyKeyOwnerOrApproved(_tokenId)
```

Comment:
The modifier should be onlyKeyManagerOrApproved, right?

Reply:
Yes! I will rework this, as it will simplify things a bit.

Reply:
For this to be safe, we need the same modifier on the approve functions

Reply:
Yes, can't let keyOwner approve themselves for example.

Reply:
I actually have to do some work on refunds. `cancelAndRefund` should have a modifier.

Reply:
What if we work on our `approve` functions so that calling `approve` (which is the expected flow from users familiar with tokens) also sets the keyManager? just thinking out loud

I definitely have to remove the owner from the existing modifier.

Reply:
Nah, gotta keep approve and setKeyManagerOf decoupled.